src/utilities.ts

```diff
@@ -73,6 +85,76 @@ export function detectChangesInWorksheet(
   return changes
 }
 
+export function detectChangesByKeyColumn(
+  prevRaw: RawWorkSheet,
+  currRaw: RawWorkSheet,
+  keyColumn: string,
+): WorkSheetChanges | undefined {
+  const changes: WorkSheetChanges = {
+    worksheetId: currRaw.worksheetId,
+    rowsRemoved: [],
+    rowsAdded: [],
+    rowsChanged: [],
+  }
+
+  if (isDetectChangesByOrder(prevRaw, currRaw, keyColumn)) {
+    // keyColumn not found, can't continue with this function
+    return detectChangesByOrder(prevRaw, currRaw)
+  }
+
+  const current = getSheetRecordFromRawWorkSheet(currRaw, keyColumn)
+  const previous = getSheetRecordFromRawWorkSheet(prevRaw, keyColumn)
+
+  Object.keys(current).map( key => {
+    if (!previous[key]) {
+      changes.rowsAdded.push(current[key])
+    }
+    else {
+      const rowChanges = detectChangesRow(previous[key], current[key])
+      if (rowChanges) {
+        changes.rowsChanged.push(rowChanges)
+      }
+    }
+  })
+
+  Object.keys(previous).map( key => {
+    if (!current[key]) {
+      changes.rowsRemoved.push(previous[key])
+    }
+  })
+  return changes
+}
+
+export function getSheetRecordFromRawWorkSheet(raw: RawWorkSheet, keyColumn: string) {
+  const record: SheetRecord = {}
+  raw.rows.forEach(function (row: any) {
+    if (row[keyColumn]) {
+      record[row[keyColumn]] = row
+    }
+  })
+  return record
+}
+
+export function isDetectChangesByOrder(prevRaw: RawWorkSheet, currRaw: RawWorkSheet, keyColumn: string) {
```

Comment:
Why is this function exported?

Also- what does it do? I think it needs to be named something else